packages/server/api/src/app/flows/flow/flow-version.controller.ts

```diff
@@ -189,6 +191,52 @@ export const flowVersionController: FastifyPluginAsyncTypebox = async (
       });
     },
   );
+
+  fastify.post(
+    '/:flowVersionId/:stepId/test',
+    TestStepRequest,
+    async (request, reply) => {
+      const { flowVersionId, stepId } = request.params;
+      const projectId = request.principal.projectId;
+
+      const flowVersion = await flowVersionService.getOneOrThrow(flowVersionId);
+
+      const step = flowHelper
+        .getAllSteps(flowVersion.trigger)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        .find((step: any) => step.id === stepId);
+
+      if (!step) {
+        await reply.status(StatusCodes.NOT_FOUND).send({
+          success: false,
+          output: 'Step not found',
+        });
+        return;
+      }
+
+      try {
+        const result = await stepRunService.create({
```

Comment:
what happens if try to execute a step and it depends on output from other steps, and that output is missing? (i.e. if you put a dependency on {{step_3}} and that was not tested?)

Reply:
hmmm, it will try to run it and for the ai the field will have 'undefined' in it, it has the same behavior as when you test it manually, it will still run but just fail